x/stakeibc/keeper/hooks_deposit_records_test.go

```diff
@@ -0,0 +1,527 @@
+		{
+			Id:                 1,
+			Amount:             0,
+			Denom:              Atom,
+			HostZoneId:         HostChainId,
+			Status:             recordstypes.DepositRecord_TRANSFER,
+			DepositEpochNumber: priorEpoch,
+		},
+		{
+			Id:                 2,
+			Amount:             0,
+			Denom:              Atom,
+			HostZoneId:         HostChainId,
+			Status:             recordstypes.DepositRecord_TRANSFER,
+			DepositEpochNumber: priorEpoch,
+		},
+	}
+
+	recordsToBeTransfered := []recordstypes.DepositRecord{
+		{
+			Id:                 3,
+			Amount:             3000,
+			Denom:              Atom,
+			HostZoneId:         HostChainId,
+			Status:             recordstypes.DepositRecord_TRANSFER,
+			DepositEpochNumber: priorEpoch,
+		},
+		{
+			Id:                 4,
+			Amount:             4000,
+			Denom:              Atom,
+			HostZoneId:         HostChainId,
+			Status:             recordstypes.DepositRecord_TRANSFER,
+			DepositEpochNumber: priorEpoch,
+		},
+	}
+	transferAmount := sdk.NewInt(3000 + 4000)
+
+	recordsToBeStaked := []recordstypes.DepositRecord{
+		{
+			Id:                 5,
+			Amount:             5000,
+			Denom:              Atom,
+			HostZoneId:         HostChainId,
+			Status:             recordstypes.DepositRecord_STAKE,
+			DepositEpochNumber: priorEpoch,
+		},
+		{
+			Id:                 6,
+			Amount:             6000,
+			Denom:              Atom,
+			HostZoneId:         HostChainId,
+			Status:             recordstypes.DepositRecord_STAKE,
+			DepositEpochNumber: priorEpoch,
+		},
+	}
+	stakeAmount := sdk.NewInt(5000 + 6000)
+
+	recordsInCurrentEpoch := []recordstypes.DepositRecord{
+		{
+			Id:                 7,
+			Amount:             7000,
+			Denom:              Atom,
+			HostZoneId:         HostChainId,
+			Status:             recordstypes.DepositRecord_STAKE,
+			DepositEpochNumber: currentEpoch,
+		},
+		{
+			Id:                 8,
+			Amount:             8000,
+			Denom:              Atom,
+			HostZoneId:         HostChainId,
+			Status:             recordstypes.DepositRecord_STAKE,
+			DepositEpochNumber: currentEpoch,
+		},
+	}
+
+	return TestDepositRecords{
+		emptyRecords:          emptyDepositRecords,
+		recordsToBeTransfered: recordsToBeTransfered,
+		recordsToBeStaked:     recordsToBeStaked,
+		recordsInCurrentEpoch: recordsInCurrentEpoch,
+		transferAmount:        transferAmount,
+		stakeAmount:           stakeAmount,
+	}
+}
+
+func (s *KeeperTestSuite) SetupDepositRecords() DepositRecordsTestCase {
+	delegationAccountOwner := fmt.Sprintf("%s.%s", HostChainId, "DELEGATION")
+	delegationChannelID := s.CreateICAChannel(delegationAccountOwner)
+	delegationAddress := s.IcaAddresses[delegationAccountOwner]
+
+	ibcDenomTrace := s.GetIBCDenom(Atom) // we need a true IBC denom here
+	hostModuleAddress := stakeibctypes.NewZoneAddress(HostChainId)
+	s.App.TransferKeeper.SetDenomTrace(s.Ctx(), ibcDenomTrace)
+
+	initialModuleAccountBalance := sdk.NewCoin(ibcDenomTrace.IBCDenom(), sdk.NewInt(15_000))
```

Comment:
where does this 15_000 come from?

Reply:
just an arbitrary number that doesn't matter much since balance changes are checked using the diff rather than the absolute value